src/som/interpreter/bc/interpreter.py

```diff
@@ -702,38 +704,43 @@ def get_self(frame, ctx_level):
 
 @elidable_promote("all")
 def _lookup(layout, selector, method, bytecode_index):
-    # First try of inline cache
-    cached_layout1 = method.get_inline_cache_layout(bytecode_index)
-    if cached_layout1 is layout:
-        invokable = method.get_inline_cache_invokable(bytecode_index)
-    elif cached_layout1 is None:
-        invokable = layout.lookup_invokable(selector)
-        method.set_inline_cache(bytecode_index, layout, invokable)
-    else:
-        # second try
-        # the bytecode index after the send is used by the selector constant,
-        # and can be used safely as another cache item
-        cached_layout2 = method.get_inline_cache_layout(bytecode_index + 1)
-        if cached_layout2 == layout:
-            invokable = method.get_inline_cache_invokable(bytecode_index + 1)
-        else:
-            invokable = layout.lookup_invokable(selector)
-            if cached_layout2 is None:
-                method.set_inline_cache(bytecode_index + 1, layout, invokable)
-    return invokable
+    cache = first = method.get_inline_cache(bytecode_index)
```

Comment:
Is there any way to avoid the code duplication between AST and BC interpreter here?

Reply:
Not sure since the code differs slightly, but I'd prefer it as well. I'll see if I can come up with something.